massa-pool-worker/src/operation_pool.rs

```diff
@@ -219,6 +221,11 @@ impl OperationPool {
                 continue;
             }
 
+            // if we have reached the maximum number of operations, stop
+            if remaining_ops == 0 {
```

Comment:
you should put this on the top of the for, to avoid continuing them with remaining_ops = 0

Reply:
Done